src/Jupyter/Extensions.cs

```diff
@@ -23,6 +23,21 @@ public static void AddIQSharpKernel(this IServiceCollection services)
         {
             services.AddSingleton<ISymbolResolver, Jupyter.SymbolResolver>();
             services.AddSingleton<IExecutionEngine, Jupyter.IQSharpEngine>();
+            services.AddSingleton<IConfigurationSource, Jupyter.IQSharpEngine>(
+                provider => provider.GetService<IExecutionEngine>() as IQSharpEngine
+            );
```

Comment:
It would be nice if we could provide default values of these configuration parameters from files or env variables.

Reply:
Agreed, my thought was to put that in a future PR, though, as there's a few more details there that I'd like to work out first if that's OK?